client/src/utils/Geometry.ts

```diff
@@ -61,8 +92,39 @@ export function calculateLinesIntersection(line1: Line, line2: Line): Dot | null
     return intersection;
 }
 
-export function eucledianDistance(dot1: Dot, dot2: Dot): number {
-    const dx = dot1.x - dot2.x;
-    const dy = dot1.y - dot2.y;
-    return Math.sqrt(dx*dx + dy*dy);
+/** Caution: Always returns an angle smaller (or equal to) 180 degrees */
+export function findAngleBetweenTwoVectors(vector1: Dot, vector2: Dot): number {
+    const dotProduct = calculateDotProduct(vector1, vector2);
+    const magnitude1 = calculateVectorMagnitude(vector1);
+    const magnitude2 = calculateVectorMagnitude(vector2);
+    const magnitudeProduct = magnitude1 * magnitude2;
+
+    return radiansToDegrees(Math.acos(dotProduct/magnitudeProduct));
+}
```

Comment:
YAGNI ([You aren't gonna need it](https://en.wikipedia.org/wiki/You_aren't_gonna_need_it)) principle... This function is never used. You avoided it with a reason (only returns angles below 180 degrees). I suggest deleting this function, especially because it's so problematic (only below 180 degrees).